GPU Health Check End-to-End tests

EddyMM opened the pull request:
### Description of changes
* End to End tests for GPU Health Checks
  * Enable Health Checks at Queue Level AND only some compute resources that have a GPU-Enabled instance type
  * Simulate a GPU Health Check failure by pointing the health check manager to a different script that returns the exit code as needed (Exit 0 to simulate successful health checks, Exit 1 otherwise)

### Tests
```
* Queue 1 (HealthChecks Enabled)
    * CR-1 (Has GPU-Enabled InstanceTypes AND Health Check Disabled)
    * CR-2 (Has GPU-Enabled InstanceTypes AND Health Check Enabled)
    * CR-3 (Has GPU-Enabled InstanceTypes AND Health Check Section NOT explicitly added)
    * CR-4 (Has NO GPU-Enabled InstanceTypes AND Health Check Disabled)
    * CR-5 (Has NO GPU-Enabled InstanceTypes AND Health Check Enabled)
    * CR-6 (Has NO GPU-Enabled InstanceTypes AND Health Check Section NOT explicitly added)
* Queue 2 (HealthChecks Not Explicitly Enabled in Config)
    * CR-1 (Has GPU-Enabled InstanceTypes AND HealthChecks Enabled)
    * CR-2 (Has NO GPU-Enabled InstanceTypes AND Health Check Enabled)
```

### References
* N/A

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

EddyMM committed:
GPU Health Check End-to-End tests

- Enable health checks at queue level and some compute resources that have a GPU-Enabled instance type then assert health checks are executed where appropriate
- Also confirms that nodes are set to drain when GPU Health Checks fail

Signed-off-by: Eddy Mwiti <eddmwiti@amazon.com>

EddyMM commented on tests/integration-tests/tests/common/schedulers_common.py:
```diff
@@ -213,6 +226,20 @@
+        """Waits until the job queue is empty."""
+
+        @retry(
+            retry_on_result=lambda result: bool(result),  # Retry internally works with only boolean values
```
[Retrying](https://pypi.org/project/retrying/) expects `retry_on_result` to be a callable function ([self._retry_on_result](https://github.com/groodt/retrying/blob/0a32064a5a6fc10777254e79b9f45d5686085c16/retrying.py#L239))

lukeseawalker commented on tests/integration-tests/tests/common/schedulers_common.py:
```diff
@@ -213,6 +226,20 @@ def _job_status_retryer():
+        """Waits until the job queue is empty."""
+
+        @retry(
+            retry_on_result=lambda result: bool(result),  # Retry internally works with only boolean values
```
NIT - for safety, strip the output

lukeseawalker commented on tests/integration-tests/tests/health_checks/test_gpu_health_checks.py:
```diff
@@ -0,0 +1,193 @@
+    compute_node_remote_command_executor = RemoteCommandExecutor(cluster, compute_node_ip=node_address)
+
+    # Mock failing GPU Health Checks
+    results_from_compute_node = compute_node_remote_command_executor.run_remote_script(failure_script_path).stdout
```
the mock and unmock can be done in the headnode. Doing this, it would probably simplify the code (e.g. no need to get compute ip address)

EddyMM committed:
- Run mock scripts from headnode since the associated directories are shared.
- Trim output from checking slurm job queue list before checking if it's empty.

Signed-off-by: Eddy Mwiti <eddmwiti@amazon.com>
